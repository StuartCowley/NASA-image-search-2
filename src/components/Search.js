import React, { useState } from "react";
import "../styles/Search.css";
import getImages from "../requests/getImages";
import propTypes from "prop-types";

const Search = ({ setSearchResults }) => {
  const [searchValue, setSearchValue] = useState();

  const handleSubmit = async (e) => {
    e.preventDefault();
    setSearchResults(await getImages(searchValue));
  };
  return (
    <>
      <form className="search__form" onSubmit={handleSubmit}>
        <input
          type="text"
          placeholder="moon"
          className="search__input"
          onChange={(e) => setSearchValue(e.target.value)}
        />
        <button type="submit" className="search__button">
          Go!
        </button>
      </form>
    </>
  );
};

Search.propTypes = {
  setSearchResults: propTypes.func,
};

export default Search;
