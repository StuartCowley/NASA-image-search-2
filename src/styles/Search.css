.search__input {
  border-radius: 11px;
  font-size: 20px;
  padding: 0 10px;
  border: 1px solid black;
}

.search__button {
  color: white;
  padding: 5px 10px;
  border-radius: 7px;
  background-color: rgb(239, 75, 75);
  border: none;
}
