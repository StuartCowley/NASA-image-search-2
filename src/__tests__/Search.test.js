import { render, screen } from "@testing-library/react";
import Search from "../components/Search";

describe("Search", () => {
  test("Search component renders correctly", () => {
    const { asFragment } = render(<Search />);

    expect(asFragment()).toMatchSnapshot();
  });
  test("Search bar renders correctly", () => {
    render(<Search />);
    const inputElement = screen.getByPlaceholderText("moon");

    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveClass("search__input");
  });
});
