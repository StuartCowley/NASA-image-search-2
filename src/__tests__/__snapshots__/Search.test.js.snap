// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Search component renders correctly 1`] = `
<DocumentFragment>
  <form
    class="search__form"
  >
    <input
      class="search__input"
      placeholder="moon"
      type="text"
    />
    <button
      class="search__button"
      type="submit"
    >
      Go!
    </button>
  </form>
</DocumentFragment>
`;
